client/my-sites/email/email-providers-stacked-comparison/provider-cards/selectors/get-product-by-interval.ts

```diff
@@ -0,0 +1,39 @@
+import 'calypso/state/products-list/init';
+import {
+	GOOGLE_WORKSPACE_BUSINESS_STARTER_MONTHLY,
+	GOOGLE_WORKSPACE_BUSINESS_STARTER_YEARLY,
+	TITAN_MAIL_MONTHLY_SLUG,
+	TITAN_MAIL_YEARLY_SLUG,
+} from '@automattic/calypso-products';
+import { IntervalLength } from 'calypso/my-sites/email/email-providers-comparison/interval-length';
+import { EmailProvider } from 'calypso/my-sites/email/form/mailboxes/types';
+import { getProductBySlug } from 'calypso/state/products-list/selectors';
+import { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Returns the correct product slug for the specified provider and interval using a map.
+ */
+const getProductSlugForInterval = ( provider: EmailProvider, intervalLength: IntervalLength ) =>
+	( {
+		[ EmailProvider.Titan ]: {
+			[ IntervalLength.MONTHLY ]: TITAN_MAIL_MONTHLY_SLUG,
+			[ IntervalLength.ANNUALLY ]: TITAN_MAIL_YEARLY_SLUG,
+		},
+		[ EmailProvider.Google ]: {
+			[ IntervalLength.MONTHLY ]: GOOGLE_WORKSPACE_BUSINESS_STARTER_MONTHLY,
+			[ IntervalLength.ANNUALLY ]: GOOGLE_WORKSPACE_BUSINESS_STARTER_YEARLY,
+		},
+	}[ provider ][ intervalLength ] );
+
+/**
+ * Retrieves the product for the specified provider and interval.
+ */
+const getProductByInterval = (
```

Comment:
Maybe the name is not that accurate? I would say that is maybe to call it `getProductByProviderAndInterval`

I was confused when I read this method when it was called. As I thought that it was going to return 2 products, as we have 2 products for each interval.